lib/ansible/modules/cloud/azure/azure_rm_subnet.py

```diff
@@ -257,6 +277,21 @@ def exec_module(self, **kwargs):
                     changed = True
                     results['route_table']['id'] = self.route_table
                     self.log("CHANGED: subnet {0} route_table to {1}".format(self.name, route_table['name']))
+
+                if self.service_endpoints:
+                    oldd = {}
+                    for item in self.service_endpoints:
+                        name = item['service']
+                        oldd[name] = {'service': name, 'locations': item['locations'].sort()}
+                    newd = {}
+                    if 'service_endpoints' in results:
+                        for item in results['service_endpoints']:
+                            name = item['service']
+                            newd[name] = {'service': name, 'locations': item['locations'].sort()}
+                    if newd != oldd:
```

Comment:
!=  operator works on dict?

Reply:
yes, tested in other modules as well

Reply:
works with nested dicts and lists

Reply:
no, it's dict newd[name] = {'service': name, 'locations': item['locations'].sort()}

Reply:
yes, dict of services

Reply:
i don't know. Pls make sure it works.